src/Plugin/Action/DataConvert.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\rules\Plugin\Action\DataConvert.
+ */
+
+namespace Drupal\rules\Plugin\Action;
+
+use Drupal\rules\Engine\RulesActionBase;
+
+/**
+ * @Action(
+ *   id = "rules_data_convert",
+ *   label = @Translation("Convert data"),
+ *   category = @Translation("Data"),
+ *   context = {
+ *     "value" = @ContextDefinition("any",
+ *       label = @Translation("Value")
+ *     ),
+ *     "target_type" = @ContextDefinition("any",
+ *       label = @Translation("Target type")
+ *     ),
+ *     "rounding_behavior" = @ContextDefinition("string",
+ *       label = @Translation("Rounding behavior"),
+ *       required = false
+ *     )
+ *   },
+ *   provides = {
+ *     "conversion_result" = @ContextDefinition("any",
+ *        label = @Translation("Conversion result")
+ *      )
+ *   }
+ * )
+ */
+class DataConvert extends RulesActionBase {
+
+  /**
+   * Executes the plugin.
+   */
+  public function execute() {
+    $value = $this->getContextValue('value');
+    $target_type = $this->getContextValue('target_type');
+    $rounding_behavior = $this->getContextValue('rounding_behavior');
+
+    // First apply the rounding behavior if given.
+    if (!empty($rounding_behavior)) {
+      switch ($rounding_behavior) {
```

Comment:
I think we should throw an exception if $rounding_behavior is not one of the values supported in the switch() statement.